### PR TITLE
lsp: set syntax instead of filetype for preview location highlighting

### LINKED ISSUE #13983: Check for valid URI before sending a `didChange` or `didOpen`.
- `nvim --version`: v0.5.0-dev+595f6e4d6
- language server name/version: client specific, the server doesn't matter.
- Operating system/version: MacOS

When previewing a location, `didChange` and `didOpen` events are being sent to the sever with an invalid uri. For example, you can test this with the following:

```
function Peek_definition()
    local params = vim.lsp.util.make_position_params()
    return vim.lsp.buf_request(0, 'textDocument/definition', params, function(_,
                                                                              _,
                                                                              res)
        vim.lsp.util.preview_location(res[1])
    end)
end
```

When the preview appears, you'll notice in the logs that at first a legit `didOpen` happens, but then a  `didChange` and `didOpen` are both then sent with `file://`. 

(_Note these logs are from the server side so swap, received and sent, I just used these since they are formatted way nicer_).

```
[Trace - 01:26:10 PM] Received request 'textDocument/definition - (3)'
Params: {
  "textDocument": {
    "uri": "file:///Users/ckipp/Documents/scala-workspace/tester/src/main/scala/example/Main.scala"
  },
  "position": {
    "line": 11,
    "character": 3
  }
}

[Trace - 01:26:10 PM] Sending response 'textDocument/definition - (3)'. Processing request took 443ms
Result: [
  {
    "uri": "file:///Users/ckipp/Documents/scala-workspace/tester/.metals/readonly/scala/Predef.scala",
    "range": {
      "start": {
        "line": 396,
        "character": 6
      },
      "end": {
        "line": 396,
        "character": 13
      }
    }
  }
]

[Trace - 01:26:10 PM] Received notification 'textDocument/didOpen'
Params: {
  "textDocument": {
    "uri": "file:///Users/ckipp/Documents/scala-workspace/tester/.metals/readonly/scala/Predef.scala",
    "languageId": "scala",
    "version": 0,
    "text": "/*\n * Scala (https://www.scala-lang.org)\n *\n * Copyright EPFL and Lightbend..."
  }
}

[Trace - 01:26:12 PM] Received notification 'textDocument/didOpen'
Params: {
  "textDocument": {
    "uri": "file://",
    "languageId": "scala",
    "version": 0,
    "text": "\n"
  }
}

[Trace - 01:26:12 PM] Received notification 'textDocument/didChange'
Params: {
  "textDocument": {
    "version": 2,
    "uri": "file://"
  },
  "contentChanges": [
    {
      "text": "   def println(x: Any) \u003d Console.println(x) \n"
    }
  ]
}

```

Just to show this isn't tied to a specific server, here is the same thing with lua

```
[ DEBUG ] 2021-02-21T13:49:44+0100 ] ...im/HEAD-595f6e4_1/share/nvim/runtime/lua/vim/lsp/rpc.lua:388 ]  "rpc.send.payload"      {  id = 4,  jsonrpc = "2.0",  method = "textDocument/definition",  params = {    position = {      character = 15,      line = 44    },    textDocument = {      uri = "file:///Users/ckipp/Documents/lua-workspace/nvim-metals/lua/metals.lua"    }  }}
[ DEBUG ] 2021-02-21T13:49:44+0100 ] ...im/HEAD-595f6e4_1/share/nvim/runtime/lua/vim/lsp/rpc.lua:489 ]  "decoded"       {  id = 4,  jsonrpc = "2.0",  result = { {      originSelectionRange = {        end = {          character = 17,          line = 44        },        start = {          character = 2,          line = 44        }      },      targetRange = {        end = {          character = 30,          line = 25        },        start = {          character = 15,          line = 25        }      },      targetSelectionRange = {        end = {          character = 30,          line = 25        },        start = {          character = 15,          line = 25        }      },      targetUri = "file:///Users/ckipp/Documents/lua-workspace/nvim-metals/lua/metals.lua"    } }}
[ DEBUG ] 2021-02-21T13:49:44+0100 ] ...im/HEAD-595f6e4_1/share/nvim/runtime/lua/vim/lsp/rpc.lua:388 ]  "rpc.send.payload"      {  jsonrpc = "2.0",  method = "textDocument/didOpen",  params = {    textDocument = {      languageId = "lua",      text = "\n",      uri = "file://",      version = 0    }  }}
[ DEBUG ] 2021-02-21T13:49:44+0100 ] ...neovim/HEAD-595f6e4_1/share/nvim/runtime/lua/vim/lsp.lua:824 ]  "on_lines bufnr: 17, changedtick: 2, firstline: 0, lastline: 1, new_lastline: 1, old_byte_size: 1, old_utf32_size: 1, old_utf16_size: 1"     { " local function execute_command(command_params, callback) " }
[ DEBUG ] 2021-02-21T13:49:44+0100 ] ...im/HEAD-595f6e4_1/share/nvim/runtime/lua/vim/lsp/rpc.lua:388 ]  "rpc.send.payload"      {  jsonrpc = "2.0",  method = "textDocument/didChange",  params = {    contentChanges = { {        text = " local function execute_command(command_params, callback) \n"      } },    textDocument = {      uri = "file://",      version = 2    }  }}
```

### Expected behaviour
I would expect for a `didChange` and `didOpen` notification *not* to be sent for an invalid uri since there is probably no reason to send it for a "scratch buffer".

### PR DESCRIPTION
**Pull request by mjlbach**
Closes #13983

TLDR we use filetype as a trigger for attaching a language server to a buffer, if we don't set filetype, then nvim-lspconfig doesn't attach the buffer (and send nil). This matches the behavior in the fancy floating markdown window.

### CODE REVIEW

**Comment by ckipp01** [comment]
Playing around with this a bit this morning and this indeed fixes it. Thanks @mjlbach!
